app/src/crypto_helper.c

```diff
@@ -0,0 +1,71 @@
+/*******************************************************************************
+ *   (c) 2018 - 2022 Zondax AG
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ ********************************************************************************/
+#include "crypto_helper.h"
+
+#include "bech32.h"
+#include "coin.h"
+#include "zxformat.h"
+#if defined(TARGET_NANOS) || defined(TARGET_NANOS2) || defined(TARGET_NANOX) || defined(TARGET_STAX)
+#include "cx.h"
+#else
+#define CX_SHA256_SIZE 32
+#define CX_RIPEMD160_SIZE 20
+#endif
+
+uint8_t bech32_hrp_len;
+char bech32_hrp[MAX_BECH32_HRP_LEN + 1];
+
+zxerr_t crypto_sha256(const uint8_t *input, uint16_t inputLen, uint8_t *output, uint16_t outputLen) {
+    if (input == NULL || output == NULL || outputLen < CX_SHA256_SIZE) {
+        return zxerr_encoding_failed;
+    }
+
+    MEMZERO(output, outputLen);
+
+#if defined(TARGET_NANOS) || defined(TARGET_NANOS2) || defined(TARGET_NANOX) || defined(TARGET_STAX)
```

Comment:
if these method are only available for real app and not cpp_test, I think that we can move this to crypto.c and we get rid of the `ifdef`